Unpin channels-redis

jessicamack opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
The bug that initially caused the upgrade block has been resolved https://github.com/django/channels_redis/issues/332.
[As a result aioredis was removed because version 4.0.0 of channel-redis migrated the underlying Redis library from aioredis to redis-py.](https://github.com/django/channels_redis/blob/13cef4502a95fbb7f41bb97eaa6f4d02c1f7c91c/CHANGELOG.txt#L37).
<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - Other

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 24.6.2
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

jessicamack committed:
unpin channels-redis

The bug that initially caused the upgrade block has been resolved https://github.com/django/channels_redis/issues/332

jessicamack committed:
replace aioredis Exception with a redis Exception

Version 4.0.0 of channel-redis migrated the underlying Redis library from aioredis to redis-py. The Exception has been changed to an equivalent

jessicamack committed:
remove unused license

jessicamack committed:
remove UPGRADE BLOCKER in README

thedoubl3j commented on requirements/requirements.in:
```diff
@@ -7,7 +7,7 @@ azure-keyvault
 botocore
 channels
-channels-redis==3.4.1  # see UPGRADE BLOCKERs
+channels-redis
```
Do we want to fully unpin this given its upgrade blocker status before?

TheRealHaoLiu replied:
its fine

TheRealHaoLiu commented on awx/main/wsrelay.py:
```diff
@@ -191,7 +191,7 @@ async def run_producer(self, name, websocket, group):
 
                     continue
-                except aioredis.errors.ConnectionClosedError:
+                except redis.exceptions.ConnectionError:
```
this might not be the right 1 to 1 replacement of the exception

jessicamack replied:
This was the exception I saw used in the aioredis code base where the ConnectionClosedError was before but I'm open to other options

TheRealHaoLiu replied:
this should be ok

jessicamack committed:
remove hiredis

it was an indirect dependency from aioredis which was removed

jessicamack committed:
remove unused license

jessicamack committed:
add back hiredis

it's potentially providing a performance boost. install explicitly as a part of redis. upgrade to more recent version

TheRealHaoLiu commented on requirements/README.md:
```diff
@@ -120,15 +120,6 @@ in the top-level Makefile.
-* https://github.com/django/channels_redis/issues/332
-* https://github.com/ansible/awx/issues/13313
-
 ### hiredis
```
This blocker is also removed in this PR

jessicamack committed:
remove UPGRADE BLOCKER for hiredis

it was also addressed as a part of this PR

jessicamack committed:
Merge branch 'devel' into upgrade-channels-redis

Sharvesh0306 commented:
@TheRealHaoLiu when will 24.6.2 be released?
